### PR TITLE
Android via azure

### PR DESCRIPTION
**Pull request by davidfurey**
Use Azure for Android notifications instead of Pushy.

I've only done basic testing for Goal Alerts, Breaking News and Content Notifications on Android so far, so I'm not quite ready to merge yet.

### CODE REVIEW

**Comment by agulowaty** [comment]
just out of interest - have you tried sending GCM notification with more than 3 tags inside using Azure ?

File: notification/app/notification/NotificationApplicationLoader.scala [hunk 1]
```diff
@@ -65,9 +66,14 @@ trait AzureHubComponents {
...
 
-  lazy val windowsNotificationSender: NotificationSender = {
+  lazy val wnsNotificationSender: WNSSender = {
     val topicSubscriptionsRepository = new DynamoTopicSubscriptionsRepository(AsyncDynamo(EU_WEST_1), appConfig.dynamoTopicsTableName)
-    new WindowsNotificationSender(hubClient, appConfig, topicSubscriptionsRepository)
+    new WNSSender(hubClient, appConfig, topicSubscriptionsRepository)
+  }
+
+  lazy val gcmNotificationSender: GCMSender = {
+    val topicSubscriptionsRepository = new DynamoTopicSubscriptionsRepository(AsyncDynamo(EU_WEST_1), appConfig.dynamoTopicsTableName)
```
Comment (agulowaty) [inline notification/app/notification/NotificationApplicationLoader.scala]: probably should be defined as component with `wire[]`

File: notification/app/notification/services/azure/NotificationsHubSender.scala [hunk 2]
```diff
@@ -64,6 +70,6 @@ class WindowsNotificationSender(hubClient: NotificationHubClient, configuration:
 }
 
 case class WindowsNotificationSenderError(underlying: Option[NotificationsError]) extends SenderError {
```
Comment (agulowaty) [inline notification/app/notification/services/azure/NotificationsHubSender.scala]: No longer specific to Windows - should be renamed

**Comment by davidfurey** [comment]
I've sent a notification with 8 topics and it worked.

File: notification/test/notification/services/azure/GCMPushConverterSpec.scala [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        Topic(FootballTeam, "29"),
+        Topic(FootballTeam, "41"),
+        Topic(FootballMatch, "3833380")
+      ),
+      addedTime = None
+    )
+
+    val azureNotification = android.GoalAlertNotification(
+      id = UUID.fromString("3e0bc788-a27c-4864-bb71-77a80aadcce4"),
+      //title = "The Guardian",
```
Comment (agulowaty) [inline notification/test/notification/services/azure/GCMPushConverterSpec.scala]: could you remove commented code ?

**Comment by agulowaty** [comment]
great one 👍

**Comment by alexduf** [comment]
Great one indeed :+1:

**Comment by davidfurey** [comment]
Still not fully regression tested on Android, but given that it won't be used by production Android apps until we add a redirect from the old tool, I think we can merge this now?

**Pull request merged**